contracts/JBveBanny.sol

```diff
@@ -239,40 +217,52 @@ contract JBveBanny is ERC721Votes, ERC721Enumerable, Ownable, ReentrancyGuard, J
   {
     if (_useJbToken) {
       // If a token wasn't set when this contract was deployed but is set now, set it.
-      if (token == IJBToken(address(0)) && tokenStore.tokenOf(projectId) != IJBToken(address(0))) {
-        token = tokenStore.tokenOf(projectId);
+      if (token == address(0) && tokenStore.tokenOf(projectId) != IJBToken(address(0))) {
+        token = address(tokenStore.tokenOf(projectId));
         // The project's token must not have changed since this token was originally set.
-      } else if (tokenStore.tokenOf(projectId) != token) revert TOKEN_MISMATCH();
+      } else if (address(tokenStore.tokenOf(projectId)) != token) revert TOKEN_MISMATCH();
     }
```

Comment:
@viraj124 could we move these checks to the constructor? Or do we need to check this each time? 

Moving them to the constructor would save about 2k gas.

Reply:
@xBA5ED The reason for doing this here is because by default we have this in constructor https://github.com/jbx-protocol/jbx-staking/blob/feat/veERC721/contracts/JBveBanny.sol#L170 & _useJbToken is passed in the lock method and it's not a part of the storage so it is dependent on user hence the reason for having this here

does that make sense ?